src/OpenLoco/TownManager.cpp

```diff
@@ -103,7 +103,8 @@ namespace OpenLoco::TownManager
             }
 
             // Write new history point.
-            currTown.history[currTown.history_size - 1] = popSteps & 0xFF;
+            auto histIndex = std::clamp(currTown.history_size - 1, 0, static_cast<int>(std::size(currTown.history)));
```

Comment:
As a final suggestion, I think the following carries the same semantics without the cast:
```suggestion
            auto histIndex = std::clamp<int>(currTown.history_size - 1, 0, std::size(currTown.history));
```

Reply:
Nice, I've completely forgotten you can use like this. Updated.